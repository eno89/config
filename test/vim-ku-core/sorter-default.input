" Test: vim-ku-core sorter-default
function s:describe__ku_sorter_default_sort()  "{{{1
  let args = {}
  let lcandidates = [
  \     {'word': 'j'},
  \     {'word': 'u'},
  \     {'word': 'n'},
  \     {'word': 'i'},
  \   ]

  It should not sort if default sorters is []

  let default_sorters = ku#sorter#default#use([])
  Should ku#sorter#default#sort(copy(lcandidates), args)
  \ ==# [{'word': 'j'}, {'word': 'u'}, {'word': 'n'}, {'word': 'i'}]

  It should sort candidates by dictionaryr order with the default sorter

  call ku#sorter#default#use(default_sorters)
  Should ku#sorter#default#sort(copy(lcandidates), args)
  \ ==# [{'word': 'i'}, {'word': 'j'}, {'word': 'n'}, {'word': 'u'}]
endfunction




function s:describe__ku_sorter_default_use()  "{{{1
  It should return the defult sorters at first

  let default_sorters = ku#sorter#default#use([])
  Should default_sorters ==# [function('ku#sorter#smart#sort')]

  It should return the previously used sorters

  let new_sorters = ku#sorter#default#use(default_sorters)
  Should new_sorters ==# []
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
