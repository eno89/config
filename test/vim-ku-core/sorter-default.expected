==== ku_sorter_default_sort 
---- It should not sort if default sorters is [] 
.
---- It requires to reset side effect 
.

==== ku_sorter_default_use 
---- It should return the defult sorters at first 
.
---- It should return the previously used sorters 
.



**** Result ****


4 examples, 0 failures
